golem-rib/src/inferred_type/unification.rs

```diff
@@ -721,10 +721,6 @@ pub fn unify_with_required(
                         inferred_type_left.clone(),
                         inferred_type_right.clone(),
                     ]))
-                } else if inferred_type_left.is_string() && inferred_type_right.is_number() {
```

Comment:
👍

Reply:
I am happy with "foo" if you give me 1 and "foo" :D 
Python inspired :P